src/server/stream_family.cc

```diff
@@ -1306,23 +1298,18 @@ OpResult<ClaimInfo> OpClaim(const OpArgs& op_args, string_view key, const ClaimO
 // XGROUP DESTROY key groupname
 OpStatus OpDestroyGroup(const OpArgs& op_args, string_view key, string_view gname) {
   auto cgr_res = FindGroup(op_args, key, gname);
-  if (!cgr_res)
-    return cgr_res.status();
-
-  if (cgr_res->cg) {
-    raxRemove(cgr_res->s->cgroups, (uint8_t*)(gname.data()), gname.size(), NULL);
-    streamFreeCG(cgr_res->cg);
+  RETURN_ON_BAD_STATUS(cgr_res);
 
-    // Awake readers blocked on this group
-    auto blocking_controller = op_args.db_cntx.ns->GetBlockingController(op_args.shard->shard_id());
-    if (blocking_controller) {
-      blocking_controller->AwakeWatched(op_args.db_cntx.db_index, key);
-    }
+  raxRemove(cgr_res->s->cgroups, (uint8_t*)(gname.data()), gname.size(), NULL);
+  streamFreeCG(cgr_res->cg);
 
-    return OpStatus::OK;
+  // Awake readers blocked on this group
+  auto blocking_controller = op_args.db_cntx.ns->GetBlockingController(op_args.shard->shard_id());
+  if (blocking_controller) {
+    blocking_controller->AwakeWatched(op_args.db_cntx.db_index, key);
   }
 
-  return OpStatus::SKIPPED;
+  return OpStatus::OK;
```

Comment:
I understand the logic has been changed here. Is it something we cover with tests?

Reply:
Reply to last 3 comments: FindGroup now return SKIPPED if the group is nullptr

So 
```c++
auto res = FindGroup();
if (!res)
  return res.status();
if (!res->cg)
  return res.status();
```

which appears multiple times turns into

```c++
auto res = FindGroup();
RETURN_ON_BAD_STATUS(res);
```